packages/webamp-modern-2/src/skin/makiClasses/WasabiFrame.ts

```diff
@@ -0,0 +1,67 @@
+import Group from "./Group";
+import UI_ROOT from "../../UIRoot";
+import { num } from "../../utils";
+
+export default class WasabiFrame extends Group {
+  __inited: boolean = false;
+  _content: string;
+  _shade: string;
+  _padtitleleft: string;
+  _padtitleright: string;
+
+  getElTag(): string {
+    return "wasabiframe";
+  }
+
+  setXmlAttr(_key: string, value: string): boolean {
+    const lowerkey = _key.toLowerCase();
+    // console.log('wasabi:frame.key=',lowerkey,':=', value)
+    if (super.setXmlAttr(lowerkey, value)) {
+      return true;
+    }
+    switch (lowerkey) {
+      case "content":
+        this._content = value;
+        break;
+      case "shade":
+        this._shade = value;
+        break;
+      case "padtitleleft":
+        this._padtitleleft = value;
+        break;
+      case "padtitleright":
+        this._padtitleright = value;
+        break;
+      default:
+        return false;
+    }
+    return true;
+  }
+
+  init() {
+    // console.error('wasabi:standard->> INITing:', this._content)
+    if (this.__inited) return;
+    this.__inited = true;
+
+    super.init();
+    // console.error('wasabi:standard->> sending onsetxuiparam:', this._content)
+    // UI_ROOT.vm.dispatch(this._systemObjects[0], "onsetxuiparam", [
+    // UI_ROOT.vm.dispatch(this, "onsetxuiparam", [
+    //   {type: "STRING", value:'content'},
+    //   {type:"STRING", value:this._content}
+    // ]);
+
+    for (const systemObject of this._systemObjects) {
+      // systemObject.init();
+      ["content", "padtitleleft", "padtitleright", "shade"].forEach((att) => {
+        const myValue = this["_" + att];
+        if (myValue != null) {
+          UI_ROOT.vm.dispatch(systemObject, "onsetxuiparam", [
```

Comment:
`std.mi` says:

>  Hookable. Event happens when the script is in a group implementing
 an xuiobject and a parameter has not been recognized by the group or
 the embedded object.
 
I think that means that these four properties are not special. Literally any XML attribute that's not otherwise recognized should trigger this.

What if we had an single property on this (or all xui) class. Something like:

```
_unhandledXuiParams: ({key: string, value: string})[];
````

Then in `setXmlAttr`, for any unrecognized key we could just do:

```
this._unhandledXuiParams.push({key: lowerkey, value});
```

And here in init we could do:

```
this._unhandledXuiParams.forEach(({key, value}) => {
  UI_ROOT.vm.dispatch(systemObject, "onsetxuiparam", [
    { type: "STRING", value: key },
     { type: "STRING", value: value },
  ]);
});
this._unhandledXuiParams = [];
```

or something like that.

---
Also, I wonder if we should have a more generic implementation of this, since it says it applies to all xui elements. I think it's okay to leave it here for now, but maybe we try to remember to pull it up.

Reply:
Perhaps we need to have more info for this. By search on all accessible modern skins, maybe?
all I did and + this your plan are = assumption.

Reply:
Yeah, we don't really know. If you prefer to leave it this way, that's fine, we can revisit once we have more info.